src/getConfig.ts

```diff
@@ -0,0 +1,21 @@
+import { exec } from 'child_process';
+
+function getConfig(): Promise<any> {
+  return new Promise((resolve, reject) => {
+    exec('npx react-native config', (error, stdout, stderr) => {
+      if (error) {
```

Comment:
hm, I feel like we have to rethink this.

This feels unstable to me. so for example if I run `npx react-native config` in my project I get the following output:

```bash
➜ npx react-native config

  WARNING: You should run npx react-native@latest to ensure you're always using the most current version of the CLI. NPX has cached version (0.73.4) != current release (0.74.2)
  
{
  "root": "/Users/hannomargelo/Documents/Github/expensify-app-fork",
```

I want to highlight that I can contain text in the stdout, which will cause the `JSON.parse` to fail.

I see the following options:

1. We add rn-cli as a _peerDependency_. This way we will have no hard dependency on it, and the rncli version of the project will be used. This way we can still use the `getConfig()` from the `/cli-config` package, which I would prefer as it also contains the type. Right now this function returns `any`, which we also want to avoid.
2. It seems we only need the context to get the package name for android. We _could_ use `exec` but then use a regex to extract the `packageName` value from the output, which seems more stable to me.

let me know what you think!

Reply:
> This feels unstable to me. so for example if I run npx react-native config in my project I get the following output:

This is how autolinking works on iOS and Android, under the hood it runs `npx react-native config` and we read stdout.

 The following output that you have is a regression that is already fixed in Core. I fixed it [here](https://github.com/facebook/react-native/pull/44644). It should be available with 0.74.2!